braindecode/datautil/windowers.py

```diff
@@ -159,7 +177,7 @@ def create_fixed_length_windows(
             # if last supercrop does not end at trial stop, make it stop
             # there
             if starts[-1] < last_allowed_ind:
-                starts[-1] = last_allowed_ind
+                starts = np.append(starts, last_allowed_ind)
```

Comment:
this is now different to before, was it wrong before?

Reply:
Good point, I forgot about this change. Can you tell me whether it does what you are expecting or not?

If my understanding is correct, what this line was doing before was **moving** the last valid window to end at the end of the recording (or at `stop_offset_samples` if specified). What the new version does is **add** a new window that ends at the end of the recording (or at `stop_offset_samples`). This way we don't loose the last valid window.

Reply:
Hm ya I went through the code as you have written it now and propose following:
Change
```python

starts = np.arange(
     start_offset_samples, stop, supercrop_stride_samples)

last_allowed_ind = stop - supercrop_size_samples
```
to
```python
last_allowed_ind = stop - supercrop_size_samples
starts = np.arange(
     start_offset_samples, last_allowed_ind+1, supercrop_stride_samples)
```
Remove
```python
if starts[-1] > last_allowed_ind:
    starts = starts[:-1]
```

Keep rest as you wrote it (so with append etc).

Reply:
With this change for me I think PR would be mergeable, maybe would try to also run some performance test myself, but could also do it post-merge

Reply:
Awesome, that's much better!

Yes, it'd be great if you could run the benchmark on your setup too, either before or after the merge.